lib/facter/java_version.rb

```diff
@@ -11,6 +11,11 @@
 #
 # Notes:
 #   None
+osfamily = Facter.value(:osfamily)
+case osfamily
+when 'OpenBSD'
+  ENV['PATH'] = ENV['PATH'] + File::PATH_SEPARATOR + '/usr/local/jdk-1.7.0/jre/bin'
```

Comment:
the java binaries are installed in that specific directory, and no links created to /usr/local/bin or /usr/local/sbin.
Ports/Packages depending on java pick up from there. therefore just adding /usr/local/bin would not help much (and it is already in the default PATH variable anyways).

I could add something like petems has proposed, implementing search_paths and which methods, where search_paths would add /usr/local/jdk-1.7.0/jre/bin to the list that ENV['PATH'] already has, and don't touch
ENV['PATH'] on others and just return what's there?

def search_paths
  

end